rust-runtime/aws-smithy-types/src/config_bag.rs

```diff
@@ -76,14 +76,17 @@ impl<T: Default> Default for Value<T> {
     }
 }
 
+/// A named layer comprising a config bag
 pub struct Layer {
```

Comment:
I think we can hide it for now. I remember having a discussion where we could pass a `Layer`, instead of `ConfigBag`, to configure interceptors to just provide a view of the `ConfigBag` to judiciously restrict an access to it on the `ConfigLoader` or `SdkConfig` level. That's what led me to keep `Layer` `pub` when I opened this PR. But if we start with private, we can always make it public later; Starting with public without concrete use cases doesn't seem like the right thing to do.

Reply:
Addressed in f55f238